src/hera/workflows/runner.py

```diff
@@ -265,7 +265,10 @@ def _runner(entrypoint: str, kwargs_list: List) -> Any:
     else:
         kwargs = _map_argo_inputs_to_function(function, kwargs)
 
-    function = validate_arguments(function)
+    # using smart union by default just in case clients do not rely on it. This means that if a function uses a union
+    # type for any of its inputs, then this will at least try to map those types correctly if the input object is
+    # not a pydantic model with smart_union enabled
+    function = validate_arguments(function, config=dict(smart_union=True))
```

Comment:
@elliotgunton also added the smart union config here, see comment

Reply:
I'm not sure about this change in terms of it being a behaviour change for users. Also I think we can do something to let users set the whole pydantic config themselves - some ideas:

* what about a `config` kwarg into the script decorator?
* Or a separate decorator e.g. `script_runner_config` that will let users set these (and more) runtime configs?
* Or moving it into the script runner constructor class and fetching it out of the script decorator?

Reply:
I'd argue this is currently limiting users, though, since the current setup can lead to all sorts of confusing scenarios such as "why am I getting a string from my union when this thing is definitely an integer" and am not sure whether the configuration would be _enabling_, so to say 🤔